ml-agents-envs/mlagents_envs/side_channel/engine_configuration_channel.py

```diff
@@ -45,32 +55,59 @@ def on_message_received(self, msg: IncomingMessage) -> None:
 
     def set_configuration_parameters(
         self,
-        width: int = 80,
-        height: int = 80,
-        quality_level: int = 1,
-        time_scale: float = 20.0,
-        target_frame_rate: int = -1,
+        width: Optional[int] = None,
+        height: Optional[int] = None,
+        quality_level: Optional[int] = None,
+        time_scale: Optional[float] = None,
+        target_frame_rate: Optional[int] = None,
+        capture_frame_rate: Optional[int] = None,
     ) -> None:
         """
         Sets the engine configuration. Takes as input the configurations of the
         engine.
-        :param width: Defines the width of the display. Default 80.
-        :param height: Defines the height of the display. Default 80.
+        :param width: Defines the width of the display. (Must be set alongside height)
+        :param height: Defines the height of the display. (Must be set alongside width)
         :param quality_level: Defines the quality level of the simulation.
-        Default 1.
         :param time_scale: Defines the multiplier for the deltatime in the
         simulation. If set to a higher value, time will pass faster in the
-        simulation but the physics might break. Default 20.
+        simulation but the physics might break.
         :param target_frame_rate: Instructs simulation to try to render at a
-        specified frame rate. Default -1.
+        specified frame rate.
+        :param capture_frame_rate: Instructs the simulation to consider time between
+        updates to always be constant, regardless of the actual frame rate.
         """
-        msg = OutgoingMessage()
-        msg.write_int32(width)
-        msg.write_int32(height)
-        msg.write_int32(quality_level)
-        msg.write_float32(time_scale)
-        msg.write_int32(target_frame_rate)
-        super().queue_message_to_send(msg)
+        if width is not None and height is not None:
```

Comment:
Since we have default values for everything in the argparser, won't these always be non-None? (at least from mlagents-learn, maybe not LL-API)

Reply:
That will be the case yes. We have defaults in learn.py.

Reply:
```
Engine Configuration:
  --width WIDTH         The width of the executable window of the
                        environment(s) in pixels (ignored for editor
                        training). (default: 84)
  --height HEIGHT       The height of the executable window of the
                        environment(s) in pixels (ignored for editor training)
                        (default: 84)
  --quality-level QUALITY_LEVEL
                        The quality level of the environment(s). Equivalent to
                        calling QualitySettings.SetQualityLevel in Unity.
                        (default: 5)
  --time-scale TIME_SCALE
                        The time scale of the Unity environment(s). Equivalent
                        to setting Time.timeScale in Unity. (default: 20)
  --target-frame-rate TARGET_FRAME_RATE
                        The target frame rate of the Unity environment(s).
                        Equivalent to setting Application.targetFrameRate in
                        Unity. (default: -1)
  --capture-frame-rate CAPTURE_FRAME_RATE
                        The capture frame rate of the Unity environment(s).
                        Equivalent to setting Time.captureFramerate in Unity.
                        (default: 60)
```